src/chart_types/partition_chart/layout/viewmodel/viewmodel.ts

```diff
@@ -248,31 +303,34 @@ export function shapeViewModel(
   // fill text
   const roomCondition = (n: ShapeTreeNode) => {
     const diff = n.x1 - n.x0;
-    return treemapLayout
+    return treemapLayout || icicleLayout || flameLayout
       ? n.x1 - n.x0 > minFontSize && n.y1px - n.y0px > minFontSize
       : (diff < 0 ? TAU + diff : diff) * ringSectorMiddleRadius(n) > Math.max(minFontSize, linkLabel.maximumSection);
   };
 
   const nodesWithRoom = quadViewModel.filter(roomCondition);
-  const outsideFillNodes = fillOutside && !treemapLayout ? nodesWithRoom : [];
+  const outsideFillNodes = fillOutside && !treemapLayout && !icicleLayout && !flameLayout ? nodesWithRoom : [];
```

Comment:
nit: Probably it's better if we change this to
```suggestion
  const outsideFillNodes = fillOutside && sunburstLayout ? nodesWithRoom : [];
```
In the case when a new layout is added, we don't have to think and remember about that in first place